kore/src/Kore/Step/EquationalStep.hs

```diff
@@ -456,3 +460,21 @@ matchRule sideCondition initial rule = do
             "Could not match patterns"
             pattern1
             pattern2
+
+evaluateRequires
+    :: forall unifier variable
+    .  SimplifierVariable variable
+    => MonadUnify unifier
+    => SideCondition variable
+    -> Condition variable
+    -> unifier (Condition variable)
+evaluateRequires sideCondition requires = do
+    _requires' <-
+        simplifyPredicate sideCondition Nothing requires
+        & withoutAxioms
```

Comment:
Why "withoutAxioms"? Is it just attempting to take a shortcut in case this can be solved without axioms? If yes, why not call `Condition.forgetSimplified` as part of `withoutAxioms`, since that seems a safer way of handling this?

Reply:
Yes, that's a good idea, with the benefit that it doesn't reset the simplified status if the first pass fails.